### PR TITLE
Add CI configuration & objdiff report

### PR DESCRIPTION
**Pull request by encounter**
Latest dtk-template changes adapted to your multi-version support

### CODE REVIEW

**Review by cadmic — COMMENTED** [review]
I see the build is not matching on main, I guess this is why we need CI

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    # Upload progress if we're on the main branch
+    - name: Upload progress
+      if: github.ref == 'refs/heads/main'
+      continue-on-error: true
+      env:
+        PROGRESS_SLUG: oot-vc
+        PROGRESS_API_KEY: ${{ secrets.PROGRESS_API_KEY }}
+      run: |
+        python tools/upload_progress.py -b https://progress.decomp.club/ \
+          -p $PROGRESS_SLUG -v oot-j build/oot-j/progress.json
```
[the remark this answers is not in the document]
Comment (Yanis42) [reply .github/workflows/build.yml:43]: I think once we support another version we can grab the changes from oot-gc

File: configure.py [hunk 2]
```diff
@@ -16,14 +16,7 @@
...
 from typing import Any, Dict, List
-
-from tools.project import (
-    Object,
-    ProjectConfig,
-    calculate_progress,
-    generate_build,
-    is_windows,
-)
+from tools.project import *
```
Comment (Yanis42) [inline configure.py:19]: I missed this before, I'm not a big fan of ``from X import *`` from my fast64 experience (which had these everywhere and caused issues)

though I guess it's not that important but I wanted to mention it

File: src/revolution/os/__start.c
Comment (Yanis42) [inline src/revolution/os/__start.c:1]: I have the entire file there btw: https://github.com/Yanis42/oot-vc/blob/rvl_sdk/src/revolution/os/__start.c

Comment (Yanis42) [reply src/revolution/os/__start.c:1]: I guess it's out of the scope of this PR though

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (encounter) [reply .github/workflows/build.yml:43]: Yep, oot-gc uses a for loop for multiple versions. I already set up oot-e and oot-u in the progress backend

File: tools/project.py [hunk 3]
```diff
@@ -265,17 +330,27 @@ def generate_build_ninja(
         deps="gcc",
     )
 
+    cargo_rule_written = False
```
[the remark this answers is not in the document]
Comment (encounter) [reply tools/project.py:333]: No danger, just a weird obsession with keeping the build.ninja simple

File: configure.py (hunk 2 above)
Comment (encounter) [reply configure.py:19]: Given that configure.py/project.py are closely tied, I wasn't too worried about this. I totally get the reasoning though.

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yanis42) [reply .github/workflows/build.yml:43]: the oot-u one has a typo in the filename I think

[the remark this answers is not in the document]
Comment (cadmic) [reply .github/workflows/build.yml:43]: what I meant is, do we have to get someone to set up a version on the backend before we can push to it here? I guess your latest comment implies yes

**Review by cadmic — APPROVED** [review]

**Review by Yanis42 — APPROVED** [review]

**Pull request merged**